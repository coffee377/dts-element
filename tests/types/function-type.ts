import * as dts from '../index';

it('should return correctly', () => {
  expect(dts.emit(
    dts.create_function_type(),
  )).toMatchSnapshot();
});

it('should return correctly with generics', () => {
  expect(dts.emit(
    dts.create_function_type({
      generics: [
        dts.create_generic_declaration({name: 'T'}),
        dts.create_generic_declaration({name: 'U'}),
      ],
    }),
  )).toMatchSnapshot();
});

it('should return correctly with parameters', () => {
  expect(dts.emit(
    dts.create_function_type({
      parameters: [
        dts.create_parameter_declaration({name: 'a'}),
        dts.create_parameter_declaration({name: 'b'}),
      ],
    }),
  )).toMatchSnapshot();
});

it('should return correctly with return', () => {
  expect(dts.emit(
    dts.create_function_type({
      return: dts.string_type,
    }),
  )).toMatchSnapshot();
});

it('should return correctly with parameters, return (TypePredicate)', () => {
  const parameter = dts.create_parameter_declaration({
    name: 'a',
  });
  expect(dts.emit(
    dts.create_function_type({
      parameters: [
        parameter,
      ],
      return: dts.create_type_predicate({
        parameter,
        type: dts.string_type,
      }),
    }),
  )).toMatchSnapshot();
});

it('should return correctly with generics, parameters, return', () => {
  expect(dts.emit(
    dts.create_function_type({
      generics: [
        dts.create_generic_declaration({name: 'T'}),
        dts.create_generic_declaration({name: 'U'}),
      ],
      parameters: [
        dts.create_parameter_declaration({name: 'a'}),
        dts.create_parameter_declaration({name: 'b'}),
      ],
      return: dts.string_type,
    }),
  )).toMatchSnapshot();
});
