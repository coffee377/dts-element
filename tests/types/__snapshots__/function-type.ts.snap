// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly 1`] = `"() => any"`;

exports[`should return correctly with generics 1`] = `"<T, U>() => any"`;

exports[`should return correctly with generics, parameters, return 1`] = `"<T, U>(a: any, b: any) => string"`;

exports[`should return correctly with parameters 1`] = `"(a: any, b: any) => any"`;

exports[`should return correctly with parameters, return (TypePredicate) 1`] = `"(a: any) => a is string"`;

exports[`should return correctly with return 1`] = `"() => string"`;
